### DISCUSSION TITLE
[v3] Form object for both a create and update form

### DISCUSSION THREAD

**Discussion by addtobag** [livewire/livewire#5922] [1. Help] [answered]
Has anyone successfully created a form object for both create and update forms?

I'm finding an error that when i'm using the form in a create i get the following error because i'm not setting $this->form->setProduct($product); in the Create Component mount function as there is no product to mount.

Cannot assign null to property App\Livewire\Forms\Product\ProductVariantForm::$product of type App\Models\Product\ProductVariant

Thanks

**Comment by joshhanley** [comment]
@addtobag What does your form object look like?

**Reply by addtobag** [reply]
Form object below, at this stage i'm just trying to dd out the form properties and this is when i'm experiencing the Cannot assign null to property error because obviously it's not set in a create form.

```
<?php

namespace App\Livewire\Forms\Product;

use Livewire\Form;
use Livewire\Attributes\Rule;
use App\Models\Product\ProductVariant;

class ProductVariantForm extends Form
{

    public ProductVariant $product;

    #[Rule('nullable')]
    public $is_active = false;

    #[Rule('nullable|min:3')]
    public $sku;

    #[Rule('nullable')]
    public $barcode;

    #[Rule('array')]
    public $options = [];


    public function setProduct(ProductVariant $product)
    {
        $this->product = $product;
    }

    public function store()
    {

        dd($this->all());

    }

    public function update()
    {
        dd($this->all());
    }
}

```

**Reply by joshhanley** [reply, marked as answer]
@addtobag just change it to this `public ?ProductVariant $product = null; `

**Reply by addtobag** [reply]
Thanks Josh, very much appreciated!

**Reply by joshhanley** [reply]
@addtobag no worries! 🙂
